lib/racecar/runner.rb

```diff
@@ -67,6 +67,8 @@ def run
       loop do
         break if @stop_requested
         resume_paused_partitions
+
+        @instrumenter.instrument("start_main_loop", instrumentation_payload)
```

Comment:
@dasch What do you think about this? I added it because it was more convenient for me but there are some alternatives.

- Add something like a 'starting' event and use that with together with 'main_loop'
- Use a listener object with `start` method to trigger at the start of the block rather than at the end

Reply:
I actually _think_ you can subscribe to the start and finish of instrumented blocks, [but it might be more trouble than it's worth](https://github.com/rails/rails/blob/7c70791470fc517deb7c640bead9f1b47efb5539/activesupport/lib/active_support/notifications/fanout.rb#L129). Basically, instead of passing a block or callable to `subscribe` you'd pass an object that responds to both `start` and `finish`.

Probably your solution is simpler :D

Reply:
Yes, that what I was saying, you have to pass an object that responds to `#start` and `#finish` which is annoying extra step.

I think the fact it wasn't obvious to either of us means we should keep the start events.